control/autoware_autonomous_emergency_braking/src/node.cpp

```diff
@@ -661,13 +692,20 @@ std::optional<Path> AEB::generateEgoPath(const Trajectory & predicted_traj)
   for (size_t i = 0; i < predicted_traj.points.size(); ++i) {
     geometry_msgs::msg::Pose map_pose;
     tf2::doTransform(predicted_traj.points.at(i).pose, map_pose, transform_stamped.value());
+
+    // skip points that are too close to the last point in the path
+    if (autoware::universe_utils::calcDistance2d(path.back(), map_pose) < 1e-2) {
+      continue;
+    }
+
     path.push_back(map_pose);
 
     if (i * mpc_prediction_time_interval_ > mpc_prediction_time_horizon_) {
       break;
     }
   }
   time_keeper_->end_track("createPath");
+  if (path.empty()) return std::nullopt;
```

Comment:
@danielsanchezaran Hello, I tested it today and things works as expected in lane_driving scenario. However in parking scenario, sometimes the trajectory will be very short and the returned path will be empty. To fix it I pushed this commit: https://github.com/autowarefoundation/autoware.universe/pull/8591/commits/f85fa019f17f225fd1fe2acc21d8f4c9d3aaae49

Reply:
@ismetatabay thank you, that is great. So I understand you tested in on the real vehicle right?

PD: I added a small recommendation regarding this line.

Reply:
@danielsanchezaran Yes, I tested it with ```speed_calculation_expansion_margin``` set to 4.0, and we didn’t stop for any overtaken NPCs during normal lane driving in a campus environment, even though we stopped for a static obstacle at 30 km/h (with -5.0 m/s² emergency deceleration and ```obstacle_cruise_planner``` disabled during the obstacle stop test). Also I pushed your suggestion :+1:  https://github.com/autowarefoundation/autoware.universe/pull/8591/commits/65f869b21d2b38d3ac5d84575a5176cd57e04c05 . If it is okay for you, I think we can merge it.

Reply:
@ismetatabay thank you, it looks great, any chance you might update the discussion about AEB with the new tests ?https://github.com/orgs/autowarefoundation/discussions/5020#discussioncomment-10312780

Also, please merge the changes!

Reply:
@danielsanchezaran Thank you as well. Our research vehicle is undergoing a hardware upgrade, and I will likely update the report next week. :+1: